### ISSUE TITLE
Hinweismeldung zu Stickstoff erscheint sobald eine Düngung gebucht wurde und noch keine Ernte eingetragen ist

### ISSUE THREAD

**Issue by simonkriegnerschramml** [w3geo/bml-fast#153]
Hallo Arnold,

Die Hinweismeldung zur Stickstoff Überdüngung kommt sobald ich eine Düngung buche und noch keinen Ertrag verbucht habe. Es scheint als ob hier der anrechenbare N dem Entzug gegenübergestellt wird.  Es sollte aber der aktiv ausgebrachte Stickstoff der errechneten Düngeobergrenze = "Düngeobergrenze nach Abzügen" (vormals "Düngeobergrenze (netto)") gegenübergestellt werden.

Bei Rückfragen bitte gerne anrufen 05069021563. Es sollte ja schnell geklärt werden können.

LG
Simon

**Comment by arnold-pichler** [comment]
OK, klar, verstehe ich. Erst sinnvoll wenn Ernte > 0. Wird umgesetzt.

**Comment by simonkriegnerschramml** [comment]
> OK, klar, verstehe ich. Erst sinnvoll wenn erw. Ertragslage > 0 (nicht Ernte). Wird umgesetzt.

Nein. Der Hinweis soll kommen, wenn die Düngung die Obergrenze nach Abzügen übersteigt.

LG
Simon

**Comment by arnold-pichler** [comment]
>  Der Hinweis soll kommen, wenn die Düngung die Obergrenze nach Abzügen übersteigt.

AH, OK. Das ist völlig unabhängig von der Ernte. OK, das muss ich umstellen, dennoch darf es erst angezeigt werden, wenn ich eine erwartete Ertragslage eingegeben habe, aktuell ist das nämlich nicht so und die Fehlermeldung erscheint auch mit "Erw. Ertragslage: Keine" - was nicht richtig sein muss.

**Comment by arnold-pichler** [comment]
Ich hatte das falsch verstanden und die Bilanz verwendet.

**Comment by simonkriegnerschramml** [comment]
> > Der Hinweis soll kommen, wenn die Düngung die Obergrenze nach Abzügen übersteigt.
> 
> AH, OK. Das ist völlig unabhängig von der Ernte. OK, das muss ich umstellen, dennoch darf es erst angezeigt werden, wenn ich eine erwartete Ertragslage eingegeben habe, aktuell ist das nämlich nicht so und die Fehlermeldung erscheint auch mit "Erw. Ertragslage: Keine" - was nicht richtig sein muss.

Genau. Der User muss, damit wir überhaupt einen Wert ausgeben können die Ertragslage angeben. Sofern das nicht geschehen ist, ist auch keine gültige Obergrenze bewertbar. Im Falle von WRRL in der Steiermark könnte man zwar schon einen Wert anzeigen sobald man die Kultur gewählt hat, aber nachdem das nicht für alle Kulturen der Fall ist und es darüber hinaus auch die Regel gibt, dass die strengere Regel gültig wird, können wir erst nach Wahl einer Ertragslage eine Düngeobergrenze berechnen.
Und die Hinweismeldung hat mit dem Ertrag nichts zu tun. Der Bauer düngt gem. einer mehrjährigen durchschnittlichen Ertragslage. Erreicht er diese in einem einzelnen Jahr nicht (Dürre, Überschwemmung, Schädlinge, etc.), heißt das nicht gleichzeitig, dass er das Gesetzt übertreten hat. Erst wenn er über seine mehrjährigen Verhältnisse hinaus düngen sollte, so wäre das eine Übertretung.

LG
Simon

**Comment by arnold-pichler** [comment]
Ist in der aktuellen Version hoffentlich korrekt umgesetzt.

**Comment by arnold-pichler** [comment]
Warnmeldungen jetzt auch für ZF

**Comment by simonkriegnerschramml** [comment]
Danke

**Comment by simonkriegnerschramml** [comment]
> Warnmeldungen jetzt auch für ZF

Die Warnmeldung erscheint jetzt auch, wenn man eine vorhandene Düngung löscht und defacto gar keine Düngung darauf liegt. Es liegt hier definitiv Problem mit dem Vorfruchtwert vor. Die Meldung erscheint, wenn ich eine Vorfrucht mit einem dementsprechend hohen Vorfruchtwert habe. Vorfruchtwerte, Nmin, N-Saldo und dergleichen sind für die Warnmeldung nicht von Relevanz. Nur die Menge an Stickstoff, die aus der Düngung stammt ist ausschlaggebend.
Die Hinweismeldung, darf auch dann nicht erscheinen, wenn die maximale N-Düngung rechnerisch negativ sein sollte. Die Hinweismeldung erscheint genau dann, wenn die maximale N-Düngung negativ wird.

Zur Lösung des Problems sollte man gar keine negativen Werte für die maximale N-Düngung zulassen. Maximal den Wert 0 anzeigen bitte.

Das kann man am Beispiel "Test8 - Gemüse auf Gemüse inkl. ZWF (1)" sehen.
[fast-export-2024-12-22 (1).txt](https://github.com/user-attachments/files/18221475/fast-export-2024-12-22.1.txt)

LG
Simon
